graphdatascience/tests/unit/test_aura_api.py

```diff
@@ -0,0 +1,186 @@
+import pytest
+from requests import HTTPError
+from requests_mock import Mocker
+
+from graphdatascience.aura_api import AuraApi, InstanceSpecificDetails
+
+
+def test_delete_instance(requests_mock: Mocker) -> None:
+    api = AuraApi(client_id="", client_secret="", tenant_id="some-tenant")
+
+    mock_auth_token(requests_mock)
+
+    requests_mock.delete(
+        "https://api.neo4j.io/v1/instances/id0",
+        status_code=202,
+        json={
+            "data": {
+                "id": "id0",
+                "name": "",
+                "status": "deleting",
+                "connection_url": "",
+                "tenant_id": "",
+                "cloud_provider": "",
+                "memory": "",
+                "region": "",
+                "type": "",
+            }
+        },
+    )
+
+    result = api.delete_instance("id0")
+
+    assert result == InstanceSpecificDetails(
+        id="id0", name="", tenant_id="", cloud_provider="", status="deleting", connection_url="", memory=""
+    )
+
+
+def test_delete_already_deleting_instance(requests_mock: Mocker) -> None:
+    api = AuraApi(client_id="", client_secret="", tenant_id="some-tenant")
+
+    mock_auth_token(requests_mock)
+    requests_mock.delete(
+        "https://api.neo4j.io/v1/instances/id0",
+        status_code=404,
+        reason="Not Found",
+        json={"errors": [{"message": "DB not found: id0", "reason": "db-not-found"}]},
+    )
+
+    result = api.delete_instance("id0")
+    assert result is None
```

Comment:
it would be nice to show to the user what the problem was

Reply:
IMO this is not a problem. They are deleting a session that's already being deleted. Things are fine.